src/components/dialogs/regulating-terminal-edition.js

```diff
@@ -0,0 +1,271 @@
+                            with enter key for it to be validated.
+                            check if autoComplete prop is fixed in v5 */}
+                    <Autocomplete
+                        size="small"
+                        freeSolo
+                        forcePopupIcon
+                        autoHighlight
+                        selectOnFocus
+                        id="equipment"
+                        disabled={
+                            !regulatingTerminalValue?.voltageLevel || disabled
+                        }
+                        options={equipmentsOptions}
+                        getOptionLabel={(equipment) => {
+                            return equipment === ''
+                                ? '' // to clear field
+                                : equipment?.type + ' : ' + equipment?.id || '';
+                        }}
+                        /* Modifies the filter option method so that when a value is directly entered in the text field, a new option
+                           is created in the options list with a value equal to the input value
+                        */
+                        filterOptions={(options, params) => {
+                            const filtered = filter(options, params);
+
+                            if (
+                                params.inputValue !== '' &&
+                                !options.find(
+                                    (opt) => opt.id === params.inputValue
+                                )
+                            ) {
+                                filtered.push({
+                                    inputValue: params.inputValue,
+                                    id: params.inputValue,
+                                });
+                            }
+                            return filtered;
+                        }}
+                        value={currentEquipment}
+                        onChange={handleChangeEquipment}
+                        renderInput={(params) => (
+                            <TextField
+                                {...params}
+                                fullWidth
+                                label={intl.formatMessage({
+                                    id: 'Equipments',
+                                })}
+                                FormHelperTextProps={{
+                                    className: classes.helperText,
+                                }}
+                            />
+                        )}
+                        PopperComponent={FittingPopper}
+                    />
+                </Grid>
+            </Grid>
+        </>
+    );
+};
+
+RegulatingTerminalEdition.propTypes = {
+    voltageLevelOptions: PropTypes.arrayOf(PropTypes.object),
+    regulatingTerminalValue: PropTypes.object,
+    onChangeVoltageLevel: PropTypes.func.isRequired,
+    onChangeEquipmentSection: PropTypes.func.isRequired,
+    direction: PropTypes.string,
```

Comment:
Missing propTypes

Reply:
doen

Reply:
Missing voltageLevelsEquipments in propTypes